dataservice/util/data_gen/data_generator.py

```diff
@@ -0,0 +1,316 @@
+            "not reported",
+            "not allowed to collect"]
+        self.gender_list = ['female', 'male', 'unknown', 'unspecified',
+                            'not reported']
+
+    def _diagnoses_choices(self):
+        """
+        Provides the choices for filling Diagnosis entity
+        """
+        self.max_diagnoses = 35
+        self.min_diagnoses = 0
+        dref_file = open('dataservice/util/data_gen/diagnoses.txt', 'r')
+        reader = csv.reader(dref_file)
+        self.diagnosis_list = []
+        for line in reader:
+            self.diagnosis_list.append(line[0])
+
+    def _genomic_files_choices(self):
+        """
+        Provides the choices for filling Genomic File entity
+        """
+        self.min_gen_files = 0
+        self.max_gen_files = 5
+        self.file_format_list = ['.cram', '.bam', '.vcf']
+
+    def setup(self, config_name):
+        """
+        creates tables in database
+        """
+        self.app = create_app(config_name)
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        db.create_all()
+
+    def teardown(self):
+        db.session.remove()
+        self.app_context.pop()
+
+    def drop_all(self):
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+    def create_and_publish_all(self):
+
+        # Create participants
+        self._create_participants(self.max_participants)
+        self.teardown()
+
+    def _create_participants(self, total):
+        """
+        creates participants with samples, demographics, and diagnoses
+        """
+        for i in range(total):
+            self.p = Participant(
+                external_id='participant_{}'.format(i),
+                samples=self._create_samples(
+                    random.randint(
+                        self.min_samples,
+                        self.max_samples)),
+                demographic=self._create_demographics(i),
+                diagnoses=self._create_diagnoses(
+                    random.randint(
+                        self.min_samples,
+                        self.max_diagnoses)))
+            db.session.add(self.p)
+        db.session.commit()
+
+    def _create_samples(self, total):
+        """
+        Create samples with aliquots
+        """
+
+        self.s_list = []
+        for i in range(total):
+            self.sample_data = {
+                'external_id': 'sample_{}'.format(i),
+                'tissue_type': random.choice(self.tissue_type_list),
+                'composition': random.choice(self.composition_list),
+                'anatomical_site': random.choice(self.anatomical_site),
+                'age_at_event_days': random.randint(0, 32872),
+                'tumor_descriptor': random.choice(self.tumor_descriptor_list)
+            }
+            self.s_list.append(
+                Sample(
+                    **self.sample_data,
+                    aliquots=self._create_aliquots(
```

Comment:
This looks pretty messy and is hard to follow. Maybe try defining the aliquots first, then passing them in?

```
aliquots = self._create_aliquots()
Sample(**sample_data, aliquots=aliquots)
```